flink-elastic-catalog/src/test/java/com/getindata/flink/connector/jdbc/catalog/ElasticCatalogTestBase.java

```
@@ -0,0 +1,130 @@
+    public static void beforeAll() throws Exception {
+        container.withEnv("xpack.security.enabled", "true");
+        container.withEnv("ELASTIC_PASSWORD", PASSWORD);
+        container.withEnv("ES_JAVA_OPTS", "-Xms1g -Xmx1g");
+        container.start();
+        TimeUnit.SECONDS.sleep(TIMEOUT_IN_SECONDS);
+        if (!isTrialEnabled()) {
+            enableTrial();
+        }
+        catalogFactory = new ElasticJdbcCatalogFactory();
+
+        url = String.format("jdbc:elasticsearch://%s:%d", container.getHost(),
+                container.getElasticPort());
+    }
+
+    private static void enableTrial() throws Exception {
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(String.format("http://%s:%d/_license/start_trial?acknowledge=true",
+                        container.getHost(), container.getElasticPort()))
+                .post(RequestBody.create(new byte[]{}))
+                .addHeader("Authorization", Credentials.basic(USERNAME, PASSWORD))
+                .build();
+        try (Response response = client.newCall(request).execute()) {
+            assertTrue(response.isSuccessful());
+        }
+    }
+
+    private static boolean isTrialEnabled() throws Exception {
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(String.format("http://%s:%d/_license",
+                        container.getHost(), container.getElasticPort()))
+                .addHeader("Authorization", Credentials.basic(USERNAME, PASSWORD))
+                .build();
+        for (int i = 0; i < REQUEST_RETRY_MAX_COUNT; i++) {
+            Response response = client.newCall(request).execute();
+            if (response.code() == 200) {
+                return response.body().string().contains("\"type\" : \"trial\"");
+            }
+            TimeUnit.SECONDS.sleep(TIMEOUT_IN_SECONDS);
+        }
+        return false;
```

If we never get 200, throw exception instead of returning false.

---

Yeah was thinking about that :)